bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/header/HeaderImpl.java

```diff
@@ -51,18 +53,43 @@
 
     private Page navigationRootPage;
 
-    @Override
-    public String getNavigationRootPageUrl() {
+    @PostConstruct
+    private void initModel() {
         if (navigationRootPage == null) {
             navigationRootPage = SiteNavigation.getNavigationRootPage(currentPage);
         }
 
         if (navigationRootPage == null) {
             LOGGER.warn("Navigation root page not found for page " + currentPage.getPath());
+        }
+    }
+
+    @Override
+    public String getNavigationRootPageUrl() {
+        if (navigationRootPage != null) {
+            return navigationRootPage.getPath() + ".html";
+        } else {
             return null;
         }
+    }
+
+    @Override
+    public String getNavigationRootPageTitle() {
+        if (navigationRootPage != null) {
+            String title = navigationRootPage.getPageTitle();
+            if (StringUtils.isNotBlank(title)) {
+                return title;
+            }
+
+            title = navigationRootPage.getTitle();
+            if (StringUtils.isNotBlank(title)) {
+                return title;
+            }
 
-        return navigationRootPage.getPath() + ".html";
+            return null;
+        } else {
```

Comment:
`else` is not needed here, just return null

Reply:
So true. :-)

Fixed both and more.